Update user guide on uploading/linking Nextcloud files from OpenProject

psatyal opened the pull request:
Added sections on:

- Uploading file to NC from OP
- Linking existing NC files from OP
- Unlinking linked files

Still WIP. Images might be added in a separate PR after implementation is merged and tested.

psatyal committed:
First draft of updated Nextcloud docs

Added sections on:

- Uploading file to NC from OP
- Linking existing NC files from OP
- Unlinking linked files

Still need to review.

psatyal committed:
Minor

psatyal commented:
Linking to WP: https://community.openproject.org/projects/openproject/work_packages/44728/activity

psatyal committed:
Added reference to drop zone

marcalcobe commented on docs/user-guide/nextcloud-integration/README.md:
```diff
@@ -69,7 +69,61 @@ In addition to listing files attached to a work package, the **Files** tab now a
+
+>> IMG: Local browser file picker
+
+Alternatively, you can also simply drag a file or folder on your computer on to this area (under the name of your Nextcloud file storage) and drop it in the drop zone that appears.
```
@psatyal maybe change "file or folder on your computer on to this area" to "file or folder from your computer on to this area"

marcalcobe commented on docs/user-guide/nextcloud-integration/README.md:
```diff
@@ -69,7 +69,61 @@ In addition to listing files attached to a work package, the **Files** tab now a
+
+Once you have selected or dropped the files you would like to upload, you will need to select a folder on Nextcloud to which they should be stored. 
+
+> **Info:** This step allows you to follow any guidelines your organisation might have concerning folder structure or file management, and where new files should be stored.
```
@psatyal I don't really understand what you mean with this info or if it is really needed.

psatyal replied:
Yeah I wanted to explain the why (as in this step allows you to follow guidlines a company might have about where to store new files), but yeah, I agree it's unnecessary. Removing.

marcalcobe commented on docs/user-guide/nextcloud-integration/README.md:
```diff
@@ -69,7 +69,61 @@ In addition to listing files attached to a work package, the **Files** tab now a
+
+To navigate one level up or to go back to the root, simply click on the relevant parent in the breadcrumb. 
+
+> **Info**: If you have navigated particularly deep (over 4 levels), intermediate levels might be collapsed to save space, but you'll always be able to navigate back to the immediate parent or the root to go backwards.
```
@psatyal do we need this info? Also we are repeating it in both sections.

psatyal replied:
Although this behaviour _should_ be obvious and easy to understand, I think it's better to mention it in the documentation, just in case it isn't. I repeat that bit on purposes, since someone might only read the bit that concerns them. (I assume users won't read all sections if there are two ways of doing the the same or a similar thing).

I'll leave it as is.

marcalcobe commented on docs/user-guide/nextcloud-integration/README.md:
```diff
@@ -69,7 +69,61 @@ In addition to listing files attached to a work package, the **Files** tab now a
+
+> **Info**: If you have navigated particularly deep (over 4 levels), intermediate levels might be collapsed to save space, but you'll always be able to navigate back to the immediate parent or the root to go backwards.
+
+Once you have selected the file(s) or folder(s) you wish to link to the work package, click on the **Link n files** button. The number on the button will represent the number of files/folders you have selected.
```
@psatyal Maybe exemplify the "N" with the symbol of number "#" would be clearer?

psatyal replied:
Good idea! Done.

marcalcobe commented on docs/user-guide/nextcloud-integration/README.md:
```diff
@@ -124,13 +188,6 @@ When a Nextcloud file or folder is linked to a work package, an OpenProject user
 
 
-
-## **Next steps for the integration**
```
@psatyal and @wielinde do we want to explain what it might come in 13.0 to the users?

psatyal replied:
I think it made sense in the last version since uploading from OP was an obvious omission that users would surely notice or wonder about. I'm removing this section because generally, documentation does not aim to inform users about upcoming changes. This feature is quite complete (from the user's pov).

psatyal committed:
Updates to the text after Marc's review

psatyal reviewed (commented):
Thank you @marcalcobe. I've made some changes (and commented where I haven't).

Hi @as-op. Could you please do a link check? It shouldn't _technically_ be necessary, but it nevers hurts to verify :)

as-op commented on docs/user-guide/nextcloud-integration/README.md:
```diff
@@ -24,10 +24,10 @@ This integration makes it possible for you to:
-| [Next steps for the integration](#next-steps-for-the-integration) | What is planned in future releases                           |
-| [Possible errors and troubleshooting](#possible-errors-and-troubleshooting) | Common errors and how to troubleshoot them                   |
+| [Linking files and folders to work packages](#linking-files-and-folders-to-work-packages) | How to link files and folders to work packages and view and download linked files |
+| [Unlinking linked files and folders](#unlinking-files-and-folders) | Unlinking linked files and folders |
```
Anchor "#unlinking-files-and-folders" should probably be "#unlinking-linked-files-and-folders" or the corresponding headline be changed to "Unlinking files and folders"

psatyal replied:
Thank you, fixed!

psatyal committed:
Updating links

Updated so that both the section header and the link correspond to "Unlinking files and folders".

psatyal commented:
Merged to dev. Will create new branch when we can add actual screenshots once the feature is implemented (and we can merge to release then).
